### PR TITLE
fix(Curriculum): ensure code that includes and is not equal to the fl…

### LINKED ISSUE #38491: Intermediate Algorithm Scripting: false negative final test
**Describe your problem and how to reproduce it:**
The final test:

> Your solution should not use the Array.prototype.flat() or Array.prototype.flatMap() methods.

fails on code that does not use either of those methods.
For example,
```
function steamrollArray(arr) {
  let results = []
  function flatten(item) {
    if(!Array.isArray(item)){
      results.push(item)
    } else {
      item.forEach(flatten)
    }
  }
  arr.forEach(flatten)
  return results
}
```
passes all the tests, except for the last one, despite not using Array.prototype.flat() or Array.prototype.flatMap() while 
```
function steamrollArray(arr) {
  let results = []
  function squish(item) {
    if(!Array.isArray(item)){
      results.push(item)
    } else {
      item.forEach(squish)
    }
  }
  arr.forEach(squish)
  return results
}
```
successfully passes all the tests, even though the only difference is that the second example uses `squish` instead of `flatten`

Additionally, all of [FCC's official solutions for this challenge](https://www.freecodecamp.org/forum/t/freecodecamp-challenge-guide-steamroller/16079) fail the last test except for solution 3, which is the only solution that does not contain 'flat' anywhere in the code

**Add a Link to the page with the problem:**
[steamroller challenge](https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/steamroller)

**Tell us about your browser and operating system:**
* Browser Name: Chrome
* Browser Version: 80.0.3987.149
* Operating System: macOS 10.15.3 

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**
Screen Shot 2020-04-02 at 1 13 18 PM

### PR DESCRIPTION
**Pull request by AshNaz87**
…at or flatMap methods do not fail the test suite

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #38491

### CODE REVIEW

**Review by hassaanp — APPROVED** [review]

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/steamroller.english.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ tests:
   - text: <code>steamrollArray([1, {}, [3, [[4]]]])</code> should return <code>[1, {}, 3, 4]</code>.
     testString: assert.deepEqual(steamrollArray([1, {}, [3, [[4]]]]), [1, {}, 3, 4]);
   - text: Your solution should not use the <code>Array.prototype.flat()</code> or <code>Array.prototype.flatMap()</code> methods.
-    testString: assert(!code.match(/\.?[\s\S]*?flat/) && !code.match(/\.?[\s\S]*?flatMap/));
+    testString: assert(!code.match(/\w+?\s*\.flat\(\)/) && !code.match(/\w+?\s*\.flatMap\(\)/));
```
Comment (moT01) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/steamroller.english.md:36]:
```suggestion
    testString: assert(!code.match(/\.\s*flat\s*\(/) && !code.match(/\.\s*flat\s*\(/));
```

**Review by moT01 — CHANGES_REQUESTED** [review]
```
function steamrollArray(arr) {
  return arr.flat(Infinity);
}
```
This would still pass, or it would pass if you use spaces.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/steamroller.english.md (hunk 1 above)
Comment (snigo) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/steamroller.english.md:36]:
```suggestion
    testString: assert(!/\.\s*flat(?:Map)?\W|\[\s*(["'`])flat(?:Map)?\1\s*\]/.test(code));
```

Will catch `.flat`, `['flat']` as well as `.flatMap` and `['flatMap']` using any combination of literal strings

**Review by snigo — CHANGES_REQUESTED** [review]

**Comment by lasjorg** [comment]
@moT01 @AshNaz87 Any update on this? We are still getting a few forum threads every now and then about it.

Latest example
https://www.freecodecamp.org/forum/t/steamroller-test-for-not-using-flat-or-flatmap-fails-yet-my-solution-does-not-use-either/377351

**Comment by AshNaz87** [comment]
@moT01 What are your thoughts?

**Review by moT01 — APPROVED** [review]
LGTM @AshNaz87 🎉 Thanks for contributing to freeCodeCamp 🎉 

@snigo has a point about being able to use bracket notation. But I doubt many will actually do that - and if they do, they're only cheating themselves. I think it's more important to just get this added so people aren't failing the challenge with code that should pass. If someone wants to make another PR with @snigo's suggestions, I would be fine with that.

**Pull request merged**